src/Generator/AbstractMemberGenerator.php

```diff
@@ -27,9 +27,9 @@ abstract class AbstractMemberGenerator extends AbstractGenerator
 
     /**
      * @param  int|int[] $flags
-     * @return AbstractMemberGenerator
+     * @return static
      */
-    public function setFlags($flags)
+    public function setFlags($flags): self
```

Comment:
Just realized that we can't add `: self` here, because child classes may be implementing `setFlags()`, and they will break, if they didn't declare `:self`

`:static` at docblock level, as well as type refinements in docblock, are sufficient

Reply:
Practical example: https://3v4l.org/f2Tu7